core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -315,62 +315,59 @@ object SparkSubmit {
       case _ =>
     }
 
-    // If we're running a python app, set the main class to our specific python runner
-    if (args.isPython && deployMode == CLIENT) {
-      if (args.primaryResource == PYSPARK_SHELL) {
-        args.mainClass = "org.apache.spark.api.python.PythonGatewayServer"
-      } else {
-        // If a python file is provided, add it to the child arguments and list of files to deploy.
-        // Usage: PythonAppRunner <main python file> <extra python files> [app arguments]
-        args.mainClass = "org.apache.spark.deploy.PythonRunner"
-        args.childArgs = ArrayBuffer(args.primaryResource, args.pyFiles) ++ args.childArgs
-        args.files = mergeFileLists(args.files, args.primaryResource)
-      }
-      args.files = mergeFileLists(args.files, args.pyFiles)
-      if (args.pyFiles != null) {
-        sysProps("spark.submit.pyFiles") = args.pyFiles
-      }
-    }
-
-    // In yarn mode for a python app, add pyspark archives to files
-    // that can be distributed with the job
-    if (args.isPython && clusterManager == YARN) {
-      var pyArchives: String = null
-      val pyArchivesEnvOpt = sys.env.get("PYSPARK_ARCHIVES_PATH")
-      if (pyArchivesEnvOpt.isDefined) {
-        pyArchives = pyArchivesEnvOpt.get
-      } else {
-        if (!sys.env.contains("SPARK_HOME")) {
-          printErrorAndExit("SPARK_HOME does not exist for python application in yarn mode.")
-        }
-        val pythonPath = new ArrayBuffer[String]
-        for (sparkHome <- sys.env.get("SPARK_HOME")) {
-          val pyLibPath = Seq(sparkHome, "python", "lib").mkString(File.separator)
-          val pyArchivesFile = new File(pyLibPath, "pyspark.zip")
-          if (!pyArchivesFile.exists()) {
-            printErrorAndExit("pyspark.zip does not exist for python application in yarn mode.")
+    if (args.isPython) {
```

Comment:
I actually find the prior organization of this code a little easier to follow. It read
```
if (args.isPython && clusterManager == YARN) { /* do something */ }
if (args.isPython && deployMode == CLIENT) { /* do something else */ }
```
So if I only cared about python but not YARN I can just skip the first block and read the second block. Now it's one giant python block. Not a big deal, but if merging these isn't strictly necessary I would prefer the old structure.